### ISSUE TITLE
Unexpeted behaviour in `counting.ipynb` notebook 

### ISSUE THREAD

**Issue by MoritzNeuberger** [scikit-hep/hepstats#139]
### Description:
When running the notebook `counting.ipynb` using `python==3.11` , `hepstats==0.7.1`, and `zfit==0.16.0`, the calculated significance of the asymptotic calculator is reported as `0.0`with a p-value of `0.5`. In addition, the rest of the notebook seems to run into a similar issue, namely that the evaluated p-value is `0.5` independent of the input.

Also, I found that the tests of the discovery calculator do not test for the significance to be >0, just <2, which might be why it slipped under the radar.

### Expected behavior: 
The significance of the discovery should 1.6 as pointed out in the markup.

### Actual Behavior:
The significance is reported as 0.0.

### Minimal working example (based on the notebook):
```
import numpy as np
import zfit
from zfit.loss import UnbinnedNLL
from zfit.minimize import Minuit
from hepstats.hypotests import Discovery, UpperLimit
from hepstats.hypotests.calculators import (AsymptoticCalculator,
                                            FrequentistCalculator)
from hepstats.hypotests.parameters import POI, POIarray

Nsig = zfit.Parameter("Nsig", 40, -100., 100)
Nbkg = zfit.Parameter("Nbkg", 340, 0, 500)
Nobs = zfit.ComposedParameter("Nobs", lambda a, b: a + b, params=[Nsig, Nbkg])

from collections import OrderedDict
import tensorflow_probability as tfp
from zfit.models.dist_tfp import WrapDistribution
from zfit.util import ztyping

class Poisson(WrapDistribution):
    _N_OBS = 1

    def __init__(self,
                 lamb: ztyping.ParamTypeInput,
                 obs: ztyping.ObsTypeInput,
                 name: str = "Poisson"):
        """
        Poisson distribution, parametrized with an event parameter (lamb).
        """
        (lamb,) = self._check_input_params(lamb)
        params = OrderedDict((('lamb', lamb),))
        dist_params = lambda: dict(rate=lamb.value())
        distribution = tfp.distributions.Poisson
        super().__init__(distribution=distribution, dist_params=dist_params,
                         obs=obs, params=params, name=name)
        
obs = zfit.Space('N', limits=(0, 800))
model = Poisson(obs=obs, lamb=Nobs) #zfit.pdf.Gauss(obs=obs, mu=Nobs, sigma=sigma)

n = 370
nbkg = 340

data = zfit.data.Data.from_numpy(obs=obs, array=np.array([n]))
Nbkg.set_value(nbkg)
Nbkg.floating = False

nll = UnbinnedNLL(model=model, data=data)
minimizer = Minuit(verbosity=0)
minimum = minimizer.minimize(loss=nll)

calculator = AsymptoticCalculator(nll, minimizer)
calculator.bestfit = minimum

discovery_test = Discovery(calculator, POI(Nsig, 0))
pnull, significance = discovery_test.result()
```
The output of this is then:
```
p_value for the Null hypothesis = 0.5
Significance (in units of sigma) = 0.0
```

### Possible Solution:
It seems that changing `fmin` to `fminfull` in `basecalculator.py:105` solves the issue.

**Comment by jonas-eschle** [comment]
Thanks a lot for reporting this issue and apologies for the delay, this went under the radar. The issue makes sense, I think the change should even be upstream and fmin should return the full likelihood (this behavior has been changed recently to make sure that if you're just accessing the likelihood, it's the full one and not the optimized one, because otherwise, things like this happen). Opened in zfit https://github.com/zfit/zfit/issues/508

Definitely worth to add a unittest for this specific example to make sure that it works. Since you've spotted it and already provide a good working example, do you want to contribute it as a test? I can gladly help you with that if you're interested to contribute

**Comment by jonas-eschle** [comment]
This is fixed in the newest 0.20.x version of zfit and hepstats will have that compatibility with #156, test added

**Comment by jonas-eschle** [comment]
released, have fun, and feedback, as always, very welcome
